### PR TITLE
Docs re-org and removal of small sections

### PR DESCRIPTION
**Pull request by weaverryan**
Hi guys!

I'd like to have less "Guides" on the left. Some of them only have 1 article or (imo) are not something that's helpful for the user. In this PR, i:

1) Move Assetic out of the top-level and under `frontend`.
2) Removed/moved the Expressions guide
3) Removed/moved the "Request" guides

On Symfony.com, we would need to remove the "Assetic", "Expressions" and "Request" guides. All of this content still exists, just somewhere else.

The bigger goal is - through a number of PR's - to reduce the number of "guides" links and make sure we have the best links that people will be looking for.

Thanks!

### CODE REVIEW

File: reference/configuration/framework.rst [hunk 1]
```diff
@@ -647,6 +649,24 @@ service
 
 This setting contains the service id of a custom matcher.
 
+request
+~~~~~~~
+
+formats
```
Comment (javiereguiluz) [inline reference/configuration/framework.rst]: Should we include here the configuration example of the original `request/mime_type.rst` article?

**Review by javiereguiluz — APPROVED** [review]
I like this a lot. All proposed changes make a lot of sense to me. Thanks!

I'll do the related changes on symfony.com.

File: reference/configuration/framework.rst (hunk 1 above)
Comment (weaverryan) [reply reference/configuration/framework.rst]: good idea - done!

**Comment by weaverryan** [comment]
@javiereguiluz I have the same question here as #8538 

> On Symfony.com, we would need to remove the "Assetic", "Expressions" and "Request" guides

Will that be automatic?

**Comment by javiereguiluz** [comment]
It's not automatic, but I've removed them in symfony.com repo, so they'll disappear after the next deployment. This is how it'd look:

tag-cloud

**Comment by weaverryan** [comment]
Thank you @javiereguiluz! It sounds like we should wait until that deploy, correct?

**Pull request closed without merging**